### ISSUE TITLE
3.11 crio_only cannot build from registry

### ISSUE THREAD

**Issue by jam01** [openshift/origin#21311]
##### Description
Cannot build from integrated registry right after clean install.
`error: build error: unable to get metadata for docker-registry.default.svc:5000/openshift/httpd@sha256:c86a751403fc65f79c155f677e07137d43dd31952134feba254791fe5f7057f6`

##### Version
```
oc v3.11.0+62803d0-1
kubernetes v1.11.0+d4cacc0
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://ip-10-0-0-9.ec2.internal:8443
openshift v3.11.0+84fe0ad-23
kubernetes v1.11.0+d4cacc0
```

##### Steps To Reproduce
1. Install OpenShift 3.11 using **deploy_cluster.yml**
2. Create an apache example app (either from web console or cli) 
cli:
`oc new-app -f https://raw.githubusercontent.com/sclorg/httpd-ex/master/openshift/templates/httpd.json`

##### Current Result
```
Cloning "https://github.com/sclorg/httpd-ex.git" ...
	Commit:	0ac6da93a1f65fe9175cb1b7838cfca7b23d5fbe (Merge pull request #15 from adambkaplan/sclorg-rename)
	Author:	Honza Horak <hhorak@redhat.com>
	Date:	Fri Aug 3 13:08:12 2018 +0200
error: build error: unable to get metadata for docker-registry.default.svc:5000/openshift/httpd@sha256:c86a751403fc65f79c155f677e07137d43dd31952134feba254791fe5f7057f6
```
##### Expected Result
Build successful

###### Verbose build log
```
oc start-build httpd-example --loglevel=10 --follow
I1019 15:46:46.771480    6427 loader.go:359] Config loaded from file /home/centos/.kube/config
I1019 15:46:46.773811    6427 loader.go:359] Config loaded from file /home/centos/.kube/config
I1019 15:46:46.777794    6427 loader.go:359] Config loaded from file /home/centos/.kube/config
I1019 15:46:46.779600    6427 loader.go:359] Config loaded from file /home/centos/.kube/config
I1019 15:46:46.783251    6427 loader.go:359] Config loaded from file /home/centos/.kube/config
I1019 15:46:46.785694    6427 request.go:897] Request Body: {"kind":"BuildRequest","apiVersion":"build.openshift.io/v1","metadata":{"name":"httpd-example","creationTimestamp":null},"triggeredBy":[{"message":"Manually triggered"}],"dockerStrategyOptions":{},"sourceStrategyOptions":{}}
I1019 15:46:46.785868    6427 round_trippers.go:386] curl -k -v -XPOST  -H "Accept: application/json, */*" -H "Content-Type: application/json" -H "User-Agent: oc/v1.11.0+d4cacc0 (linux/amd64) kubernetes/d4cacc0" 'https://ip-10-0-0-9.ec2.internal:8443/apis/build.openshift.io/v1/namespaces/apache-test/buildconfigs/httpd-example/instantiate'
I1019 15:46:47.193402    6427 round_trippers.go:405] POST https://ip-10-0-0-9.ec2.internal:8443/apis/build.openshift.io/v1/namespaces/apache-test/buildconfigs/httpd-example/instantiate 201 Created in 407 milliseconds
I1019 15:46:47.193428    6427 round_trippers.go:411] Response Headers:
I1019 15:46:47.193435    6427 round_trippers.go:414]     Cache-Control: no-store
I1019 15:46:47.193439    6427 round_trippers.go:414]     Content-Type: application/json
I1019 15:46:47.193444    6427 round_trippers.go:414]     Content-Length: 1443
I1019 15:46:47.193448    6427 round_trippers.go:414]     Date: Fri, 19 Oct 2018 15:46:47 GMT
I1019 15:46:47.193840    6427 request.go:897] Response Body: {"kind":"Build","apiVersion":"build.openshift.io/v1","metadata":{"name":"httpd-example-3","namespace":"apache-test","selfLink":"/apis/build.openshift.io/v1/namespaces/apache-test/buildconfigs/httpd-example-3/instantiate","uid":"2fa0e1df-d3b6-11e8-85e8-0669e8a5098a","resourceVersion":"78729","creationTimestamp":"2018-10-19T15:46:47Z","labels":{"app":"httpd-example","buildconfig":"httpd-example","openshift.io/build-config.name":"httpd-example","openshift.io/build.start-policy":"Serial","template":"httpd-example"},"annotations":{"openshift.io/build-config.name":"httpd-example","openshift.io/build.number":"3"},"ownerReferences":[{"apiVersion":"build.openshift.io/v1","kind":"BuildConfig","name":"httpd-example","uid":"4a375069-d3b2-11e8-85e8-0669e8a5098a","controller":true}]},"spec":{"serviceAccount":"builder","source":{"type":"Git","git":{"uri":"https://github.com/sclorg/httpd-ex.git"}},"strategy":{"type":"Source","sourceStrategy":{"from":{"kind":"DockerImage","name":"docker-registry.default.svc:5000/openshift/httpd@sha256:c86a751403fc65f79c155f677e07137d43dd31952134feba254791fe5f7057f6"},"pullSecret":{"name":"builder-dockercfg-dbw64"}}},"output":{"to":{"kind":"ImageStreamTag","name":"httpd-example:latest"}},"resources":{},"postCommit":{},"nodeSelector":null,"triggeredBy":[{"message":"Manually triggered"}]},"status":{"phase":"New","config":{"kind":"BuildConfig","namespace":"apache-test","name":"httpd-example"},"output":{}}}
build.build.openshift.io/httpd-example-3 started
I1019 15:46:47.197133    6427 round_trippers.go:386] curl -k -v -XGET  -H "User-Agent: oc/v1.11.0+d4cacc0 (linux/amd64) kubernetes/d4cacc0" -H "Accept: application/json, */*" 'https://ip-10-0-0-9.ec2.internal:8443/apis/build.openshift.io/v1/namespaces/apache-test/builds/httpd-example-3/log?follow=true'
I1019 15:46:50.029078    6427 round_trippers.go:405] GET https://ip-10-0-0-9.ec2.internal:8443/apis/build.openshift.io/v1/namespaces/apache-test/builds/httpd-example-3/log?follow=true 200 OK in 2831 milliseconds
I1019 15:46:50.029102    6427 round_trippers.go:411] Response Headers:
I1019 15:46:50.029108    6427 round_trippers.go:414]     Cache-Control: no-store
I1019 15:46:50.029113    6427 round_trippers.go:414]     Content-Type: application/octet-stream
I1019 15:46:50.029117    6427 round_trippers.go:414]     Date: Fri, 19 Oct 2018 15:46:50 GMT
Cloning "https://github.com/sclorg/httpd-ex.git" ...
	Commit:	0ac6da93a1f65fe9175cb1b7838cfca7b23d5fbe (Merge pull request #15 from adambkaplan/sclorg-rename)
	Author:	Honza Horak <hhorak@redhat.com>
	Date:	Fri Aug 3 13:08:12 2018 +0200
error: build error: unable to get metadata for docker-registry.default.svc:5000/openshift/httpd@sha256:c86a751403fc65f79c155f677e07137d43dd31952134feba254791fe5f7057f6
```

###### oc adm diagnostics
https://pastebin.com/uD940qCz
###### oc get all -o json
https://pastebin.com/tcJP6pKP

###### Related to https://github.com/openshift/origin/issues/8372 ?
These other issues seems similar however I'm not using any Proxies or docker at all, so I'm not sure if that's even useful.

**Comment by ekrunch** [comment]
Just a heads up, I was having the same problem and I found a glitch in my installation with /etc/containers/registries.conf that was causing the same problems. I'm on 3.11 Enterprise with the CRI-O option enabled and using CRI-O node profiles in my inventory.

Check your /etc/containers/registries.conf and look at the "[registries.insecure]" section. If your "registries" entry is [""], that's invalid and will cause Docker to fail on startup. I edited that file to the following

**[registries.insecure]
registries = []**

And then started up Docker. Everything worked great after this adjustment.

Hope this helps.
Ed
